src/commands/ImportObjectsCommandGenerator.js

```diff
@@ -29,102 +29,75 @@ const ANSWERS_NAMES = {
-								SDKOperationResultUtils.logResultMessage(operationResult);
-								SDKOperationResultUtils.logErrors(operationResult);
-								return;
-							}
-							if (Array.isArray(data) && operationResult.data.length === 0) {
-								NodeUtils.println(
-									TranslationService.getMessage(MESSAGES.NO_OBJECTS_TO_LIST),
-									NodeUtils.COLORS.RESULT
-								);
-								return;
-							}
-
-							const questions = this._generateSelectionObjectQuestions(
-								operationResult
-							);
-
-							prompt(questions).then(secondAnswers => {
-								const combinedAnswers = { ...firstAnswers, ...secondAnswers };
-								const finalAnswers = this._arrangeAnswersForImportObjects(
-									combinedAnswers
-								);
-								resolve(finalAnswers);
-							});
-						})
-						.catch(error => {
-							reject(
-								TranslationService.getMessage(
-									ERRORS.CALLING_LIST_OBJECTS,
-									NodeUtils.lineBreak,
-									error
-								)
-							);
-						});
-				})
-				.catch(error =>
-					reject(
-						TranslationService.getMessage(
-							PROMPTING_INTERACTIVE_QUESTIONS_FAILED,
-							NodeUtils.lineBreak,
-							error
-						)
-					)
-				);
-		});
+	async _getCommandQuestions(prompt) {
+		try {
+			const listObjectQuestions = this._generateListObjectQuestions();
+			const listObjectAnswers = await prompt(listObjectQuestions);
+
+			const paramsForListObjects = this._arrangeAnswersForListObjects(listObjectAnswers);
+			const executionContextForListObjects = new SDKExecutionContext({
+				command: this._listObjectsMetadata.name,
+				params: paramsForListObjects,
+				includeProjectDefaultAuthId: true,
+			});
+
+			let operationResult;
+			try {
+				operationResult = await executeWithSpinner({
+					action: this._sdkExecutor.execute(executionContextForListObjects),
+					message: TranslationService.getMessage(MESSAGES.LOADING_OBJECTS),
+				});
+			} catch (error) {
+				return Promise.reject(TranslationService.getMessage(ERRORS.CALLING_LIST_OBJECTS, NodeUtils.lineBreak, error));
+			}
+
+			const { data } = operationResult;
+			if (SDKOperationResultUtils.hasErrors(operationResult)) {
+				SDKOperationResultUtils.logResultMessage(operationResult);
+				SDKOperationResultUtils.logErrors(operationResult);
+				return;
+			}
+			if (Array.isArray(data) && operationResult.data.length === 0) {
+				NodeUtils.println(TranslationService.getMessage(MESSAGES.NO_OBJECTS_TO_LIST), NodeUtils.COLORS.RESULT);
+				return;
+			}
+
+			const selectionObjectQuestions = this._generateSelectionObjectQuestions(operationResult);
+
+			const selectionObjectAnswers = await prompt(selectionObjectQuestions);
+			const combinedAnswers = { ...listObjectAnswers, ...selectionObjectAnswers };
+			const finalAnswers = this._arrangeAnswersForImportObjects(combinedAnswers);
+			return finalAnswers;
+		} catch (error) {
+			return Promise.reject(TranslationService.getMessage(PROMPTING_INTERACTIVE_QUESTIONS_FAILED, NodeUtils.lineBreak, error));
```

Comment:
I don't think we need "Promise.reject" here. Just rethrowing should be enough:
```
 throw TranslationService.getMessage(PROMPTING_INTERACTIVE_QUESTIONS_FAILED, NodeUtils.lineBreak, error); 
```

Reply:
Change made.